Tests/Marketplace/marketplace_services.py

```diff
@@ -3007,7 +3026,212 @@ def copy_author_image(self, production_bucket, build_bucket, images_data, storag
+
+                for image_info in readme_images_storage_paths:
+                    readme_original_url = image_info.get('original_read_me_url')
+                    gcs_storage_path = str(image_info.get('new_gcs_image_path'))
+                    image_name = str(image_info.get('image_name'))
+
+                    task_status = self.download_readme_image_from_url_and_upload_to_gcs(readme_original_url,
+                                                                                        gcs_storage_path,
+                                                                                        image_name, storage_bucket)
+                    self._reademe_images.append(image_name)
+
+        except Exception:
+            logging.exception(f"Failed uploading {self._pack_name} pack readme image.")
+            task_status = False
+        finally:
+            return task_status
+
+    def collect_images_from_readme_and_replace_with_storage_path(self, pack_readme_path, gcs_pack_path, marketplace):
+        """
+        Replaces inplace all images links in the pack README.md with their new gcs location
+
+        Args:
+            pack_readme_path (str): A path to the pack README file.
+            gcs_pack_path (str): A path to the pack in gcs
+            marketplace (str): The marketplace this pack is going to be uploaded to.
+
+        Returns:
+            A list of dicts of all the image urls found in the README.md file with all related data
+            (original_url, new_gcs_path, image_name)
+        """
+        logging.info(f'{marketplace=}')
+        if marketplace == 'xsoar':
+            marketplace = 'marketplace-dist'
+        else:
+            marketplace = 'marketplace-v2-dist'
+
+        google_api_readme_images_url = f'https://storage.googleapis.com/{marketplace}/content/packs/{self.name}'
+        url_regex = r"^!\[(.*)\]\((?P<url>.*)\)"
+        urls_list = []
+
+        for line in fileinput.input(pack_readme_path, inplace=True):
+            res = re.search(url_regex, line)
+            # we found a matching url and we want to modify the readme line.
+            if res:
+                url = res.group('url')
+
+                parse_url = urllib.parse.urlparse(url)
+                path = parse_url.path
+                url_path = Path(path)
+                image_name = url_path.name
+
+                image_gcp_path = Path(gcs_pack_path, BucketUploadFlow.README_IMAGES, image_name)
+                url_to_replace_linking_to_dist = os.path.join(google_api_readme_images_url,
+                                                              BucketUploadFlow.README_IMAGES, image_name)
+
+                line = line.replace(url, str(url_to_replace_linking_to_dist))
+
+                urls_list.append({
+                    'original_read_me_url': url,
+                    'new_gcs_image_path': image_gcp_path,
+                    'image_name': image_name
+                })
+
+            print(line, end='')
```

Comment:
Remove the print, add a log instead

Reply:
This is not a regular print the fileinput module will redirect the stdout to edit the readme file inplace.